src/simulations/optimisation.py

```diff
@@ -124,11 +136,24 @@ def _car_efficiency_iterations(self, prm, res):
 
         return res
 
-    def _power_flow_equations(self, p, netp, grid, hourly_line_losses_pu):
+    def _power_flow_equations(
+            self, p, netp, grid, hourly_line_losses_pu,
+            charge, discharge_other, totcons):
         # loads on network from agents
         voltage_costs = p.add_variable('voltage_costs', 1)  # daily voltage violation costs
         pi = p.add_variable('pi', (self.grd['n_buses'] - 1, self.N), vtype='continuous')
-        netq = p.add_variable('netq', (self.n_homes, self.N), vtype='continuous')
+        # variables only needed for passive homes and reactive power
+        if self.n_homesP > 0:
+            q_heat_home_car_non_flex = p.add_variable(
```

Comment:
I think you use the extensions `_non_flex` and `_passive` interchangeably throughout the code?
It would be good to clarify. I think I prefer `_passive` to be used everywhere?